leetcode/src/69.c

```diff
@@ -0,0 +1,24 @@
+//using binary search method is one of the efficient one for this problem statement.
+int mySqrt(int x){
+int start=0;
+        int end=x;
+        long long int ans=0;
+        while(start <= end){
+               long long int mid=(start+end)/2;
+               if(mid*mid == x){
+                         ans=mid;
+                         break;
+                }
+//if mid is less than the square root of the number(x) store the value of mid in ans.
+               if( mid*mid < x){
+                         ans = mid;
+                       start = mid+1;
+               }
+//if mid is greater than the square root of the number(x) then ssign the value mid-1 to end.               
+               if( mid*mid > x){
```

Comment:
You are doing multiplication 3 times per loop in bad case. You can store the  result of `mid * mid` in variable and use it.

Reply:
ok thanks for the suggestion i will consider that.